### PR TITLE
Add PostgreSQLQueue and DelayedPostgreSQL Pub/Sub

### PR DESCRIPTION
**Pull request by m110**
* Added `PostgreSQLQueueSchema` and `PostgreSQLQueueOffsetsAdapter`. 
    * This schema is similar to the original one, except it allows filtering messages with a custom `WHERE` clause.
    * It doesn't support customer groups.
    * It allows deleting messages from the table after acking (optional).
* Added `NewDelayedPostgreSQLPublisher` and `NewDelayedPostgresSQLSubscriber` 
    * They work on top of the conditional schema and use the `delay` component implemented in https://github.com/ThreeDotsLabs/watermill/pull/469
    * The idea is to receive messages with a delay or at a given time in a simple way.
* Reworked the `SchemaAdapter` and `OffsetsAdapter` interfaces **(Breaking change -> will bump the major to v4)**.
    * We had a few instances where some details have been missing from one of the interface methods. It makes it difficult to extend this library with new features, as each will require a major version bump, and we want to avoid it if possible.
    * To mitigate this, we're moving to a `params` struct passed to the methods instead of raw arguments. It adds some verbosity but allows extending the params if needed with no breaking changes to the library.
    * The methods now also return errors.
    * Migration: if you don't use a custom schema/offsets adapter, you don't need to do anything. If you do, change the methods to implement the new interfaces. Use values from the `params` argument and return errors.

### CODE REVIEW

File: pkg/sql/conditional_schema_adapter_postgresql.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+	// GenerateMessagesTableName may be used to override how the messages table name is generated.
+	GenerateMessagesTableName func(topic string) string
+
+	// SubscribeBatchSize is the number of messages to be queried at once.
+	//
+	// Higher value, increases a chance of message re-delivery in case of crash or networking issues.
+	// 1 is the safest value, but it may have a negative impact on performance when consuming a lot of messages.
+	//
+	// Default value is 100.
```
[the remark this answers is not in the document]
Comment (m110) [reply pkg/sql/conditional_schema_adapter_postgresql.go:36]: As discussed, let's keep it compatible with other schemas

File: pkg/sql/conditional_offsets_adapter_postgresql.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package sql
+
+import (
+	"fmt"
+
+	"github.com/lib/pq"
+)
+
+// ConditionalPostgreSQLOffsetsAdapter is an OffsetsAdapter for the ConditionalPostgreSQLSchema.
+type ConditionalPostgreSQLOffsetsAdapter struct {
```
[the remark this answers is not in the document]
Comment (m110) [reply pkg/sql/conditional_offsets_adapter_postgresql.go]: c611c62eee2cbefccd80eb614f652c70eeb926f4

File: pkg/sql/conditional_offsets_adapter_postgresql.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func (a ConditionalPostgreSQLOffsetsAdapter) SchemaInitializingQueries(params OffsetsSchemaInitializingQueriesParams) []Query {
+	return []Query{}
+}
+
+func (a ConditionalPostgreSQLOffsetsAdapter) NextOffsetQuery(params NextOffsetQueryParams) Query {
+	return Query{}
+}
+
+func (a ConditionalPostgreSQLOffsetsAdapter) AckMessageQuery(params AckMessageQueryParams) Query {
+	if params.ConsumerGroup != "" {
```
[the remark this answers is not in the document]
Comment (m110) [reply pkg/sql/conditional_offsets_adapter_postgresql.go]: ed30aa87f43f9076e8ee7a1629d58412ffb4b70b

**Review by roblaszczak — APPROVED** [review]

File: pkg/sql/queue_offsets_adapter_postgresql.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package sql
+
+import (
+	"fmt"
+
+	"github.com/lib/pq"
+)
+
+// PostgreSQLQueueOffsetsAdapter is an OffsetsAdapter for the PostgreSQLQueueSchema.
+type PostgreSQLQueueOffsetsAdapter struct {
```
Comment (roblaszczak) [inline pkg/sql/queue_offsets_adapter_postgresql.go:10]: I guess it would be good to add it to watermill.io - leaving to not forget

Comment (m110) [reply pkg/sql/queue_offsets_adapter_postgresql.go:10]: I plan to, but wanted to merge this one first so the docs src-link works.

**Pull request merged**